### ISSUE TITLE
Update is performed two times while closing an window

### ISSUE THREAD

**Issue by MatthijsPrent** [elmish/Elmish.WPF#540]
**High-level description**
First of all, thank you for creating Elmish.Wpf, I like using it for my f# WPF application. I have a why my update is executed twice with the same message, and what I should do to prevent this.

**Steps to Reproduce**
I have created a very simple project to demonstrate my situation:
https://github.com/MatthijsPrent/ElmishNewWindowTest

This application should open and close a secondary window. However, it does not exactly behave like I would suppose it would. When I put an breakpoint at the CloseWindow step in my update method, and close the secondary window, I can see that the breakpoint is hit twice, where I would expect it to be hit only once. I would like to know why this happens, and what I can do to fix this. I am aware of the new window example project, It is different from mine, but I don't see the key difference.

**Additional Information**
Elmish.WPF version: 4.
Target framework version: Core 6.

**Comment by TysonMN** [comment]
I found two cases for the `NewWindow` sample.  Closing via the `X` button dispatches twice and closing via the `Submit` button only dispatches once.

![2022-10-28_07-03-41_722_devenv](https://user-images.githubusercontent.com/34664007/198583904-dd8deb38-c03c-4e9b-b87d-d8a6db980df8.gif)

Thanks very much for providing a reproduction.  That allowed me to share this video your program with you.

I split your `CloseWindow` message in two.  Then, if I close via the `X` button, there are two dispatches of the `CloseWindow1` message, just like the `NewWindow` sample.  But if I close via your `Close` button, then there is one dispatch of the `CloseWindow2` message followed by one dispatch of the `CloseWindow1` message.

![2022-10-28_07-08-00_728_devenv](https://user-images.githubusercontent.com/34664007/198584468-3fa291c1-429a-47b7-8648-151413ac8eb1.gif)

Is this enough of a clue figure out how to get your program to behave how you want?
